comp/core/settings/settingsimpl/settingsimpl.go

```diff
@@ -0,0 +1,221 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package settingsimpl implements the interface for the settings component
+package settingsimpl
+
+import (
+	"encoding/json"
+	"html"
+	"net/http"
+	"sync"
+
+	"go.uber.org/fx"
+	"gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/settings"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newSettings),
+	)
+}
+
+type provides struct {
+	fx.Out
+
+	Comp settings.Component
+}
+
+type dependencies struct {
+	fx.In
+
+	Log      log.Component
+	Settings []settings.RuntimeSetting `group:"runtime_setting"`
+}
+
+type settingsRegistry struct {
+	rwMutex            sync.RWMutex
+	registeredSettings map[string]settings.RuntimeSetting
+	log                log.Component
+}
+
+// RuntimeSettings returns all runtime configurable settings
+func (s *settingsRegistry) RuntimeSettings() map[string]settings.RuntimeSetting {
+	return s.registeredSettings
+}
+
+// GetRuntimeSetting returns the value of a runtime configurable setting
+func (s *settingsRegistry) GetRuntimeSetting(setting string) (interface{}, error) {
+	s.rwMutex.RLock()
+	defer s.rwMutex.RUnlock()
+	if _, ok := s.registeredSettings[setting]; !ok {
+		return nil, &settings.SettingNotFoundError{Name: setting}
+	}
+	value, err := s.registeredSettings[setting].Get()
+	if err != nil {
+		return nil, err
+	}
+	return value, nil
+}
+
+// SetRuntimeSetting changes the value of a runtime configurable setting
+func (s *settingsRegistry) SetRuntimeSetting(setting string, value interface{}, source model.Source) error {
+	s.rwMutex.Lock()
+	defer s.rwMutex.Unlock()
+	if _, ok := s.registeredSettings[setting]; !ok {
+		return &settings.SettingNotFoundError{Name: setting}
+	}
+	return s.registeredSettings[setting].Set(value, source)
+}
+
+func (s *settingsRegistry) GetFullConfig(cfg config.Config, namespaces ...string) http.HandlerFunc {
```

Comment:
I don't fully understand what is this function doing in `settingsRegistry`. It doesn't seems to use the data from `settingsRegistry` and seems like it should be part of the config comp or something similar.

Reply:
I copied over because it was defined in the settings server struct

https://github.com/DataDog/datadog-agent/blob/20b6bac12ec3ad6534edb85a359a7804df3e93e3/pkg/config/settings/http/server.go#L25

I agree that this one belongs more in the configuration component. To avoid creating too much change I say to leave in the settings component to recreate as much as possible the setting package and once we can register routes move it to the config component.